src/types/acls.rs

```diff
@@ -56,16 +56,32 @@ impl From<Vec<AclRight>> for AclRightList {
     }
 }
 
-impl From<&str> for AclRightList {
-    fn from(i: &str) -> Self {
-        i.chars()
-            .into_iter()
+impl TryFrom<&str> for AclRightList {
+    type Error = AclRightError;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        if !input
+            .chars()
+            .all(|c| c.is_ascii_lowercase() || c.is_ascii_digit())
+        {
+            return Err(AclRightError::InvalidRight);
+        }
+
+        Ok(input
+            .chars()
             .map(|c| c.into())
             .collect::<HashSet<AclRight>>()
-            .into()
+            .into())
     }
 }
 
+/// Error from parsing AclRight strings
+#[derive(Debug, Eq, PartialEq)]
+pub enum AclRightError {
```

Comment:
We'll definitely want to implement `std::error::Error` for this. I'm also not sure it needs to be an `enum`? If we truly expect new variants might be added in the future though, this should be marked `#[non_exhaustive]`.

Reply:
I added non_exhausted and implemented Error.  And by using an enum it makes it easier to extend in the future w/o breaking API.